lib/r10k/puppetfile.rb

```diff
@@ -212,15 +221,22 @@ def concurrent_accept(visitor, pool_size)
   def modules_queue(visitor)
     Queue.new.tap do |queue|
       visitor.visit(:puppetfile, self) do
-        modules.each { |mod| queue << mod }
+        modules_by_cachedir = modules_by_vcs_cachedir.clone
+        modules_without_vcs_cachedir = modules_by_cachedir.delete(:none) || []
+
+        modules_without_vcs_cachedir.each {|mod| queue << Array(mod) }
```

Comment:
I'm a confused by this.. `modules_without_vcs_cachedir` could only be `:none` or `[]`, right? so why would those need to go in the queue?

Reply:
`:none` isn't going in the queue, `:none` is a key containing all the modules that don't have a cachedir. They're added one-by-one to queue so that they can all get processed in parallel.

Reply:
oh right it's a map, not an array

Reply:
so then wouldn't we want the values of `{none: [<blah]}`, instead of the whole map? like we do for `modules_by_cachedir` below?

Reply:
maybe I should remind myself how Ruby works

Reply:
![ignore me](https://i.pinimg.com/originals/39/6b/17/396b17ee8c2664e3bc72b4e70bb62cc7.gif)